packages/react/src/components/utils/attachEventProps.ts

```diff
@@ -19,7 +19,8 @@ export const attachEventProps = (node: HTMLElement, newProps: any, oldProps: any
         syncEvent(node, eventNameLc, newProps[name]);
       }
     } else {
-      if (typeof newProps[name] === 'object') {
+      const propType = typeof newProps[name];
+      if (propType === 'object' || propType === 'undefined') {
         (node as any)[name] = newProps[name];
       } else {
         node.setAttribute(camelToDashCase(name), newProps[name]);
```

Comment:
Do we need to setAttribute at all? convert a prop name to dash case might not  work, we should always set property:
```
if (name in node) {
    node[name] = newProps[name];
}
```

Also, if our code adds attributes, we need to also add code to  remove them (I don't think we should do that at all), we should use properties 99%  of the times. In fact, we remove pure CSS attributes for this reason, also JSX  will complain about any non-prop passed

Reply:
On a side note, why is this function called `attachEventProps` seems like it's doing a lot more than that